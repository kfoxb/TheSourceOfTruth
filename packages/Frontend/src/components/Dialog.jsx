import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import MDialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogTitle from '@material-ui/core/DialogTitle';

export default function Dialog(props) {
  const {
    dialogIsOpen,
    handleClose,
    handleSubmit,
  } = props;

  return (
    <div>
      <MDialog
        open={dialogIsOpen}
        onClose={handleClose}
        aria-labelledby="responsive-dialog-title"
      >
        <DialogTitle id="responsive-dialog-title">Are you sure you want to submit?</DialogTitle>
        <DialogActions>
          <Button onClick={handleClose} color="primary">
              Cancel
          </Button>
          <Button onClick={handleSubmit} color="primary" autoFocus>
              Submit
          </Button>
        </DialogActions>
      </MDialog>
    </div>
  );
}

Dialog.propTypes = {
  dialogIsOpen: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};
