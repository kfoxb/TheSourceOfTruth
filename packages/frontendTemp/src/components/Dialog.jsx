import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import MDialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import colors from '../constants/colors';

export default function Dialog({
  buttons,
  dialogIsOpen,
  content,
  handleClose,
  title,
}) {
  return (
    <div>
      <MDialog
        open={dialogIsOpen}
        onClose={handleClose}
        aria-labelledby="responsive-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="responsive-dialog-title">
          {title}
        </DialogTitle>
        { content &&
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              {content}
            </DialogContentText>
          </DialogContent>
        }
        <DialogActions>
          {buttons.map(button =>
            (React.isValidElement(button) ? button : (
              <Button key={button.label} onClick={button.action} style={{ color: `${colors.blue}` }}>
                {button.label}
              </Button>)))}
        </DialogActions>
      </MDialog>
    </div>
  );
}

Dialog.propTypes = {
  buttons: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      action: PropTypes.func.isRequired,
    }).isRequired,
    PropTypes.element.isRequired,
  ]).isRequired).isRequired,
  content: PropTypes.string,
  dialogIsOpen: PropTypes.bool.isRequired,
  handleClose: PropTypes.func,
  title: PropTypes.string.isRequired,
};

Dialog.defaultProps = {
  content: '',
  handleClose: () => {},
};
