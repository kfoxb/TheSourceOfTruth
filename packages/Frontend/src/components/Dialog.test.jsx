/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import Dialog from './Dialog';

describe('Dialog Component', () => {
  it('should be a function', () => {
    expect(typeof Dialog).toBe('function');
  });

  it('should render Dialog when dialogIsOpen is true', () => {
    const wrapper = shallow(<Dialog
      dialogIsOpen
      handleClose={() => {}}
      handleSubmit={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render Dialog when dialogIsOpen is false', () => {
    const wrapper = shallow(<Dialog
      dialogIsOpen={false}
      handleClose={() => {}}
      handleSubmit={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
