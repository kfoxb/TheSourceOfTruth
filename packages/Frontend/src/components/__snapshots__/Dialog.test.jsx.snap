// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component should not render Dialog when dialogIsOpen is false 1`] = `
<div>
  <WithStyles(Dialog)
    aria-describedby="alert-dialog-description"
    aria-labelledby="responsive-dialog-title"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogTitle)
      id="responsive-dialog-title"
    />
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        onClick={[Function]}
        style={
          Object {
            "color": "#039BE5",
          }
        }
      />
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`Dialog Component should render Dialog when dialogIsOpen is true 1`] = `
<div>
  <WithStyles(Dialog)
    aria-describedby="alert-dialog-description"
    aria-labelledby="responsive-dialog-title"
    onClose={[Function]}
    open={true}
  >
    <WithStyles(DialogTitle)
      id="responsive-dialog-title"
    />
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        onClick={[Function]}
        style={
          Object {
            "color": "#039BE5",
          }
        }
      />
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;
